pkg/operator/ceph/cluster/osd/config.go

```diff
@@ -106,3 +116,41 @@ func (c *Cluster) skipVolumeForDirectory(path string) bool {
 	// since the dataDirHostPath is always mounting at /var/lib/rook
 	return path == k8sutil.DataDir
 }
+
+func encryptionKeyPath() string {
+	return fmt.Sprintf("%s/%s", opconfig.EtcCephDir, encryptionKeyFileName)
+}
+
+func encryptionDMName(pvcName string) string {
+	return fmt.Sprintf("%s-block-dmcrypt", pvcName)
+}
+
+func encryptionDMPath(pvcName string) string {
+	return fmt.Sprintf("/dev/mapper/%s", encryptionDMName(pvcName))
+}
+
+func generateDmCryptKey() (string, error) {
+	key, err := mgr.GenerateRandomBytes(dmCryptKeySize)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate random bytes")
+	}
+
+	return base64.StdEncoding.EncodeToString(key), nil
+}
+
+func generateOSDEncryptedKeySecret(pvcName, namespace, key string) *v1.Secret {
+	return &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
```

Comment:
How about adding a label with the pvc name?

Reply:
Sounds good.

Reply:
We don't know of a need for it today, but it's possible it will be useful. In a follow-up PR is fine too.